packages/n4s/src/enforce/compounds/oneOf.js

```diff
@@ -0,0 +1,14 @@
+import one from 'one';
+import runLazyRules from 'runLazyRules';
+import { withFirst } from 'withArgs';
+
+/**
+ * @param {*} value   Value to be test against rules
+ * @param {Function[]} rules    Rules to validate the value with
+ */
+function oneOf(value, rules) {
+  return one.apply(null,
```

Comment:
>It could be a bit wasteful and keep running even after more than two tests passed. I would have preferred to short circuit immediately as we detect more than one successful test.
Maybe we could try something similar to this?

I may have mislead you with my previous comment. This works, but it adds some unneeded internal call to the `run` module function which performs some extra checks that we do not use here.

What I meant is that you implement similar logic, but not require the file itself. If you find it easily extractable, you could extract it to some shared location, but that's not a priority.